diff -ruN -x __pycache__ -x obj -x '*~' -x tags -x TAGS ParallelIO-pio2_5_10.orig/CMakeLists.txt ParallelIO-pio2_5_10/CMakeLists.txt
--- ParallelIO-pio2_5_10.orig/CMakeLists.txt	2022-12-10 15:35:33.000000000 +0100
+++ ParallelIO-pio2_5_10/CMakeLists.txt	2022-12-12 16:31:07.957314633 +0100
@@ -4,7 +4,7 @@
 
 # Jim Edwards
 
-cmake_minimum_required (VERSION 3.5.2)
+cmake_minimum_required (VERSION 3.10)
 project (PIO C)
 
 # The project version number.
@@ -146,17 +146,12 @@
       WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
       OUTPUT_QUIET
       ERROR_QUIET)
-    find_path (USER_CMAKE_MODULE_PATH
-      NAMES mpiexec.cmake
-      HINTS ${CMAKE_BINARY_DIR}/CMake_Fortran_utils)
-    if (USER_CMAKE_MODULE_PATH)
-      message (STATUS "Importing CMake_Fortran_utils - success")
-    else ()
-      message (FATAL_ERROR "Failed to import CMake_Fortran_utils")
-    endif ()
+    set (USER_CMAKE_MODULE_PATH "${CMAKE_BINARY_DIR}/CMake_Fortran_utils"
+      CACHE STRING "Location of the CMake_Fortran_utils")
+  else ()
+    set (USER_CMAKE_MODULE_PATH ${USER_CMAKE_MODULE_PATH}
+      CACHE STRING "Location of the CMake_Fortran_utils")
   endif ()
-  set (USER_CMAKE_MODULE_PATH ${USER_CMAKE_MODULE_PATH}
-  CACHE STRING "Location of the CMake_Fortran_utils")
   list (APPEND CMAKE_MODULE_PATH ${USER_CMAKE_MODULE_PATH})
 endif ()
 
diff -ruN -x __pycache__ -x obj -x '*~' -x tags -x TAGS ParallelIO-pio2_5_10.orig/src/clib/CMakeLists.txt ParallelIO-pio2_5_10/src/clib/CMakeLists.txt
--- ParallelIO-pio2_5_10.orig/src/clib/CMakeLists.txt	2022-12-10 15:35:33.000000000 +0100
+++ ParallelIO-pio2_5_10/src/clib/CMakeLists.txt	2022-12-12 16:28:44.197280881 +0100
@@ -171,6 +171,9 @@
     set (WITH_PNETCDF FALSE)
     set (MPI_C_INCLUDE_PATH ${MPISERIAL_C_INCLUDE_DIRS})
   endif ()
+else ()
+  target_link_libraries (pioc
+    PUBLIC MPI::MPI_C)
 endif ()
 
 include(CheckTypeSize)
diff -ruN -x __pycache__ -x obj -x '*~' -x tags -x TAGS ParallelIO-pio2_5_10.orig/src/flib/CMakeLists.txt ParallelIO-pio2_5_10/src/flib/CMakeLists.txt
--- ParallelIO-pio2_5_10.orig/src/flib/CMakeLists.txt	2022-12-10 15:35:33.000000000 +0100
+++ ParallelIO-pio2_5_10/src/flib/CMakeLists.txt	2022-12-12 16:28:44.198280888 +0100
@@ -158,6 +158,9 @@
     set (WITH_PNETCDF FALSE)
     set (MPI_Fortran_INCLUDE_PATH ${MPISERIAL_Fortran_INCLUDE_DIRS})
   endif ()
+else ()
+    target_link_libraries (piof
+      PUBLIC MPI::MPI_Fortran)
 endif ()
 
 # Check MPI I/O capabilities
diff -ruN -x __pycache__ -x obj -x '*~' -x tags -x TAGS ParallelIO-pio2_5_10.orig/src/gptl/CMakeLists.txt ParallelIO-pio2_5_10/src/gptl/CMakeLists.txt
--- ParallelIO-pio2_5_10.orig/src/gptl/CMakeLists.txt	2022-12-10 15:35:33.000000000 +0100
+++ ParallelIO-pio2_5_10/src/gptl/CMakeLists.txt	2022-12-12 16:28:44.200280903 +0100
@@ -112,6 +112,8 @@
   if (MPI_C_FOUND AND MPI_Fortran_FOUND)
     target_compile_definitions (gptl
       PUBLIC HAVE_MPI)
+    target_link_libraries (gptl
+      PUBLIC MPI::MPI_C MPI::MPI_Fortran)
   endif ()
 endif ()
 
