diff -ruN -x __pycache__ -x obj -x '*~' -x tags -x TAGS ParallelIO-pio2_5_9.orig/CMakeLists.txt ParallelIO-pio2_5_9/CMakeLists.txt
--- ParallelIO-pio2_5_9.orig/CMakeLists.txt	2022-09-19 20:47:16.000000000 +0200
+++ ParallelIO-pio2_5_9/CMakeLists.txt	2022-11-25 15:40:48.237696077 +0100
@@ -4,7 +4,7 @@
 
 # Jim Edwards
 
-cmake_minimum_required (VERSION 3.7)
+cmake_minimum_required (VERSION 3.10)
 project (PIO C)
 
 # The project version number.
@@ -143,20 +143,15 @@
     message (STATUS "Importing CMake_Fortran_utils")
     execute_process(
       COMMAND git clone https://github.com/CESM-Development/CMake_Fortran_utils
-      WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+      WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
       OUTPUT_QUIET
       ERROR_QUIET)
-    find_path (USER_CMAKE_MODULE_PATH
-      NAMES mpiexec.cmake
-      HINTS ${CMAKE_BINARY_DIR}/CMake_Fortran_utils)
-    if (USER_CMAKE_MODULE_PATH)
-      message (STATUS "Importing CMake_Fortran_utils - success")
-    else ()
-      message (FATAL_ERROR "Failed to import CMake_Fortran_utils")
-    endif ()
+    set (USER_CMAKE_MODULE_PATH "${CMAKE_BINARY_DIR}/CMake_Fortran_utils"
+      CACHE STRING "Location of the CMake_Fortran_utils")
+  else ()
+    set (USER_CMAKE_MODULE_PATH ${USER_CMAKE_MODULE_PATH}
+      CACHE STRING "Location of the CMake_Fortran_utils")
   endif ()
-  set (USER_CMAKE_MODULE_PATH ${USER_CMAKE_MODULE_PATH}
-  CACHE STRING "Location of the CMake_Fortran_utils")
   list (APPEND CMAKE_MODULE_PATH ${USER_CMAKE_MODULE_PATH})
 endif ()
 
diff -ruN -x __pycache__ -x obj -x '*~' -x tags -x TAGS ParallelIO-pio2_5_9.orig/src/clib/CMakeLists.txt ParallelIO-pio2_5_9/src/clib/CMakeLists.txt
--- ParallelIO-pio2_5_9.orig/src/clib/CMakeLists.txt	2022-09-19 20:47:16.000000000 +0200
+++ ParallelIO-pio2_5_9/src/clib/CMakeLists.txt	2022-11-25 14:20:02.718666935 +0100
@@ -171,6 +171,9 @@
     set (WITH_PNETCDF FALSE)
     set (MPI_C_INCLUDE_PATH ${MPISERIAL_C_INCLUDE_DIRS})
   endif ()
+else ()
+  target_link_libraries (pioc
+    PUBLIC MPI::MPI_C)
 endif ()
 
 include(CheckTypeSize)
diff -ruN -x __pycache__ -x obj -x '*~' -x tags -x TAGS ParallelIO-pio2_5_9.orig/src/flib/CMakeLists.txt ParallelIO-pio2_5_9/src/flib/CMakeLists.txt
--- ParallelIO-pio2_5_9.orig/src/flib/CMakeLists.txt	2022-09-19 20:47:16.000000000 +0200
+++ ParallelIO-pio2_5_9/src/flib/CMakeLists.txt	2022-11-25 14:26:48.086534925 +0100
@@ -158,6 +158,9 @@
     set (WITH_PNETCDF FALSE)
     set (MPI_Fortran_INCLUDE_PATH ${MPISERIAL_Fortran_INCLUDE_DIRS})
   endif ()
+else ()
+    target_link_libraries (piof
+      PUBLIC MPI::MPI_Fortran)
 endif ()
 
 # Check MPI I/O capabilities
diff -ruN -x __pycache__ -x obj -x '*~' -x tags -x TAGS ParallelIO-pio2_5_9.orig/src/gptl/CMakeLists.txt ParallelIO-pio2_5_9/src/gptl/CMakeLists.txt
--- ParallelIO-pio2_5_9.orig/src/gptl/CMakeLists.txt	2022-09-19 20:47:16.000000000 +0200
+++ ParallelIO-pio2_5_9/src/gptl/CMakeLists.txt	2022-11-25 15:42:22.747330593 +0100
@@ -112,6 +112,8 @@
   if (MPI_C_FOUND AND MPI_Fortran_FOUND)
     target_compile_definitions (gptl
       PUBLIC HAVE_MPI)
+    target_link_libraries (gptl
+      PUBLIC MPI::MPI_C MPI::MPI_Fortran)
   endif ()
 endif ()
 
